### DISCUSSION TITLE
what time format should be used?

### DISCUSSION THREAD

**Discussion by modpotato** [pocketbase/pocketbase#5439] [Q&A] [answered]
i had heard somewhere to use RFC3339 but 
```go
func main() {
	fmt.Println(time.Now().Format(time.RFC3339))
}
```
returns 2009-11-10T23:00:00Z, and when i send a request i get [list] can't unmarshal response, err parsing time "2024-08-30 01:18:48.280Z" as "2006-01-02T15:04:05Z07:00": cannot parse " 01:18:48.280Z" as "T"

though i am using an unofficial api (github.com/pluja/pocketbase), and ive opened an issue for that, i just want to be sure im formatting it properly

the request in question:
```go
	existingArchives, err := userArchiveCollection.List(pocketbase.ParamsList{
		Filters: fmt.Sprintf("artist_id.id = '%s' && created > '%s'", userID, lastModified.Format(time.RFC3339)),
		Sort:    "-created",
		Size:    1,
	})
```

im fairly new to go so i have a feeling im just doing something wrong.. am i just using the wrong formatting? or is this a problem with their api?

**Comment by ganigeorgiev** [comment, marked as answer]
RFC3399 has several formats and a datetime with space instead of `T` is one of them.

The full PocketBase stored datetime format in Go layout time format is `2006-01-02 15:04:05.000Z`.

_It would depend on your use case and what you are trying to do but just to clear some eventual misunderstanding - if you are using Go already you can [use PocketBase directly as a regular Go package](https://pocketbase.io/docs/go-overview/) instead of sending HTTP requests. For working with the PocketBase dates there is also [`types.DateTime`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.20/tools/types) helper (with the next release it will support all `time.Time` methods)._

Support for other date formats could be added eventually in the future but that will be after the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355).

**Reply by modpotato** [reply]
yeah, i think i need to fork the library im using as its trying to format as rfc3399 with the T
also, i dont intend to embed pocketbase into my app as theres no built-in compression yet which is quite the deal breaker. thanks so much though

side note, it may have been a problem with the struct i was using as i was defining Created as below
```go
	Created         time.Time              `json:"created"`
```
removing this and just using sorting and limits seems to work fine

**Reply by ganigeorgiev** [reply]
> theres no built-in compression

I don't understand what do you mean by this and I'll not pursue the matter further, but again PocketBase is distributed as a regular Go package and you have direct access to the database and its internals. Sending HTTP requests to it from another Go app is fine but it will be always less efficient than direct access (you also can't benefit from the hooks).

**Reply by modpotato** [reply]
oh im writing a web scraper worker, just seems inefficient to spin up a new database for each worker but im open to suggestion :)

**Reply by ganigeorgiev** [reply]
I'm sorry, I don't understand what do you mean by _"spin up a new database for each worker"_. I don't get it why users refuse to elaborate more clearly on their use cases but I'm getting tired by the constant back-and-forth and will stop engage further.

As a last attempt to avoid eventual misunderstanding, you can start the PocketBase server in a seperate go routine as part of your existing program, aka. you can wrap in `go func() { ... }` the app.Start() and once the application is bootstrapped you can operate directly with the running `app` instance using the available Go APIs - https://pocketbase.io/docs/go-overview/. This will be even simpler with the upcoming [next release](https://github.com/pocketbase/pocketbase/discussions/4355) (_there are no ETAs yet_). The main file of the example prebuilt executable could be found in https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go.

**Reply by modpotato** [reply]
i see what you mean, what i was trying to say is the application im writing in go is meant to be ran in a distributed network all pointing back to a database (pocketbase for now) or will in the future (i dont want to block off that possibility, as it is a webscraper and ideally can horizontally scale) especially because i want to make the pocketbase instance (theoretically inside the application) public as i intend to make a frontend for it at some point which may run in an object store and not directly on a host. so in the meantime im using pockethost and will eventually migrate to a cloudflare tunneled vm out of concern for ddos attacks, which is the main motivating factor behind keeping pocketbase isolated. and my other concern is my current hardware setup doesnt have much storage not already dedicated to vms and because of that i would prefer to run pocketbase on a host for now until i can.
i had kept my responses short as i had already gotten an answer and wasnt sure if i should continue talking about it, but i assume this is fine..
sorry for any confusion
